ihatemoney/models.py

```diff
@@ -346,7 +346,10 @@ def pay_each(self):
         """Compute what each share has to pay"""
         if self.owers:
             # FIXME: SQL might dot that more efficiently
-            return self.amount / sum(i.weight for i in self.owers)
+            weights = sum(i.weight for i in self.owers)
+            if weights <= 0:
```

Comment:
If I understand correctly, this can happen if, before this commit is merged, somebody had negative weights. If so, wouldn't it make more sense to:

- keep this code as is
- migrate former negative weights to 1?

Then we have the general invariant that weights are non-negative.